packages/cells/src/model.ts

```diff
@@ -424,7 +424,9 @@ export class RawCellModel extends AttachmentsCellModel {
    * Serialize the model to JSON.
    */
   toJSON(): nbformat.IRawCell {
-    return super.toJSON() as nbformat.IRawCell;
+    const cell = super.toJSON() as nbformat.IRawCell;
+    cell.id = this.id;
```

Comment:
Test:
- [ ] a markdown cell preserves its ids in its toJSON output
- [ ] a raw cell preserves its ids in its toJSON output
- [ ] a code cell preserves its ids in its toJSON output